### PR TITLE
feat: add description for rich link previews

### PR DESCRIPTION
**Pull request by LunrEclipse**
Updates meta tags for NFT assets and collections to include information regarding the NFT collection. Also added description tags to tokens including price and ticket.

### CODE REVIEW

File: functions/utils/types.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import { gql } from '@apollo/client'
```
Comment (zzmp) [inline functions/utils/types.ts:1]: Why is this necessary? Is it because you're making a new query, but the types are generated from src? Does that mean you were just re-using queries from src?

You should generate your own query here if possible. If not possible, you should commit a query into src (with comments as to why it lives there) and then pull it from there so you get auto-magic typings.

wdyt?

**Review by zzmp — CHANGES_REQUESTED** [review]

File: functions/utils/types.ts (hunk 1 above)
Comment (LunrEclipse) [reply functions/utils/types.ts:1]: yeah i was just re-using queries from src... 

I think its probably best to generate my own queries since like 90% of the information from src aren't needed so we can strip out a lot and really simply it. should i do that in this pr or a new one?

Comment (zzmp) [reply functions/utils/types.ts:1]: if you _can_ do that in this PR it would be great - if you've got time :)
